### PR TITLE
Gnosis interface tweaks

### PR DESCRIPTION
**Pull request by evgenyboxer**
# Description

Interface tweaks and lots of bug fixes - based on https://www.figma.com/file/FnVmt8Ho1rdwgk26fyNDtE/TradeUI?node-id=337%3A12822

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Make multiple trades via Balancer/Gnosis (ETH->ERC20 and ERC20<>ERC20)

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where relevant, particularly in hard-to-understand areas
- [x] My changes generate no new console warnings
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/plugins/blocknative.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export const defaultOptions: InitializationOptions = {
   dappId: process.env.VUE_APP_BLOCKNATIVE_DAPP_ID || '',
   networkId: Number(process.env.VUE_APP_NETWORK) || 1,
   onerror: error => {
-    console.log(`[Blocknative] encountered an error - ${error}`);
+    console.log(`[Blocknative] encountered an error`, error);
```
Comment (evgenyboxer) [inline src/plugins/blocknative.ts:10]: This is not related to the PR but needed to show proper errors

File: src/locales/default.json [hunk 2]
```diff
@@ -264,10 +275,26 @@
...
+            "header": "Insufficient {0} balance",
+            "body": "The fees and potential slippage (up to {2}) set for this trade exceeds your {0} balance. You need at least {1} {0} to proceed."
+        }
+    },
+    "gnosisWarnings": {
+        "highFees": {
+            "header": "The fees on this transaction exceed 20% of the trade amount.",
+            "body": "You can still proceed with the trade but a high percentage of it will be consumed by trade fees."
+        }
+    },
```
Comment (evgenyboxer) [inline src/locales/default.json:297]: I think this pattern of keys is much more manageable and is not dependent on the actual content.

File: src/composables/trade/useGnosis.ts [hunk 3]
```diff
@@ -23,7 +23,17 @@ import useNumbers from '../useNumbers';
...
+const HIGH_FEE_THRESHOLD = 0.2;
+const INITIAL_STATE = {
+  errors: {
+    feeExceedsPrice: false,
+    priceExceedsBalance: false
+  },
+  warnings: {
+    highFees: false
+  }
+};
```
Comment (garethfuller) [inline src/composables/trade/useGnosis.ts:36]: Just define initial state outside the composable function and then use it internally.
```suggestion
const state = reactive({
  errors: {
    feeExceedsPrice: false,
    priceExceedsBalance: false
  },
  warnings: {
    highFees: false
  }
});
```
Then in the composable below you can update like:
```js
state.errors.feeExceedsPrice = true;
```
Then to export:
```js
return {
  ...toRefs(state),
}
```

File: src/composables/trade/useGnosis.ts [hunk 4]
```diff
@@ -72,16 +84,11 @@ export default function useGnosis({
 
   // DATA
   const feeQuote = ref<FeeInformation | null>(null);
-  const errors = ref({
-    feeExceedsPrice: false
-  });
+  const errors = ref(INITIAL_STATE.errors);
+  const warnings = ref(INITIAL_STATE.warnings);
```
Comment (garethfuller) [inline src/composables/trade/useGnosis.ts:88]: You shouldn't need to do this if you use the state pattern above.

File: src/composables/trade/useSor.ts [hunk 5]
```diff
@@ -32,6 +32,12 @@ const GAS_PRICE = process.env.VUE_APP_GAS_PRICE || '100000000000';
 const MAX_POOLS = process.env.VUE_APP_MAX_POOLS || '4';
 const SWAP_COST = process.env.VUE_APP_SWAP_COST || '100000';
 const MIN_PRICE_IMPACT = 0.0001;
+const HIGH_PRICE_IMPACT_THRESHOLD = 0.05;
+const INITIAL_STATE = {
+  errors: {
+    highPriceImpact: false
+  }
+};
```
Comment (garethfuller) [inline src/composables/trade/useSor.ts:40]: Again you could be using the reactive state pattern here, but you could also actually include all the composable state in one reactive object.

**Review by garethfuller — APPROVED** [review]
Looks good, just some comments about composable state patterns.

File: src/composables/trade/useSor.ts (hunk 5 above)
Comment (evgenyboxer) [reply src/composables/trade/useSor.ts]: Yeah good call!

**Pull request merged**